### PR TITLE
Add thomson to nstx machine mappings and fix ECH for DIIID IMAS v3.37.0

### PR DESCRIPTION
**Pull request by jmcclena**

### CODE REVIEW

File: omas/machine_mappings/nstxu.py [hunk 1]
```diff
@@ -492,6 +492,48 @@ def mse_data(ods, pulse, MSE_revision="ANALYSIS", MSE_Er_correction=True):
             coef_list.get(f'AA{IMAS2GAM[k]}GAM', [0] * (ch + 1))[ch] for k in range(9)
         ]
 
+# ================================
+@machine_mapping_function(__regression_arguments__, pulse=140001)
+def thomson_scattering_hardware(ods, pulse):
+    """
+    Gathers DIII-D Thomson measurement locations
```
Comment (orso82) [inline omas/machine_mappings/nstxu.py:499]: DIII-D --> NSTX(-U) here and elsewhere
```suggestion
    Gathers NSTX(-U) Thomson measurement locations
```

File: omas/machine_mappings/nstxu.py [hunk 2]
```diff
@@ -492,6 +492,48 @@ def mse_data(ods, pulse, MSE_revision="ANALYSIS", MSE_Er_correction=True):
...
+    """
+    Loads DIII-D Thomson measurement data
+
+    :param pulse: int
+    """
+    
+
+    signals = ['FIT_NE', 'FIT_NE_ERR','FIT_TE','FIT_TE_ERR','FIT_RADII','TS_TIMES']
+    signals_norm = {'FIT_NE':1.0,'FIT_NE_ERR':1.0,'FIT_TE':1e3,'FIT_TE_ERR':1e3,'FIT_RADII':1e-2,'TS_TIMES':1.0}
+
```
Comment (orso82) [inline omas/machine_mappings/nstxu.py:518]: you can format the OMAS source by typing `make format` at the top-level folder

File: omas/machine_mappings/nstxu.py [hunk 3]
```diff
@@ -492,6 +492,48 @@ def mse_data(ods, pulse, MSE_revision="ANALYSIS", MSE_Er_correction=True):
...
+    TDIs = {}
+    for item in signals:
+        TDI = f'\\ACTIVESPEC::TOP.MPTS.OUTPUT_DATA.BEST.{item}'
+        TDIs[item] = '\\'+TDI.strip('\\')
+    res = mdsvalue('nstxu', pulse=pulse, treename='NSTX', TDI=TDIs).raw()
+    print(res)
```
Comment (orso82) [inline omas/machine_mappings/nstxu.py:524]: cruft

**Review by orso82 — CHANGES_REQUESTED** [review]
looks good. Only few minor changes

**Comment by smithsp** [comment]
@jmcclena has addressed changes.  Requested review by @orso82  again.

**Comment by orso82** [comment]
@jmcclena if you find it useful, these are the updated ECH mappings as I did them https://github.com/gafusion/omas/tree/ech_imasv37

**Comment by orso82** [comment]
@jmcclena let me know when/if you want to merge

**Comment by smithsp** [comment]
https://github.com/gafusion/omas/compare/jmcclena_nstx_thomson...ech_imasv37 may be useful

**Pull request merged**